src/hidimstat/base_variable_importance.py

```diff
@@ -0,0 +1,131 @@
+import warnings
+
+from sklearn.base import BaseEstimator
+import numpy as np
+
+
+class BaseVariableImportance(BaseEstimator):
+    """
+    Base class for variable importance methods.
+
+    This class provides a foundation for implementing variable importance methods,
+    including feature selection based on importance scores and p-values.
+
+    Attributes
+    ----------
+    importances_ : array-like of shape (n_features,), default=None
+        The computed importance scores for each feature.
+    pvalues_ : array-like of shape (n_features,), default=None
+        The computed p-values for each feature.
+    selections_ : array-like of shape (n_features,), default=None
+        Binary mask indicating selected features.
+
+    Methods
+    -------
+    selection(k_best=None, percentile=None, threshold=None, threshold_pvalue=None)
+        Selects features based on importance scores and/or p-values using various criteria.
+
+    _check_importance()
+        Checks if importance scores and p-values have been computed.
+    """
+
+    def __init__(self):
+        super().__init__()
+        self.importances_ = None
+        self.pvalues_ = None
+        self.selections_ = None
+
+    def selection(
+        self, k_best=None, percentile=None, threshold=None, threshold_pvalue=None
+    ):
+        """
+        Selects features based on variable importance.
```

Comment:
```suggestion
        Selects features based on variable importance.
        In case several arguments are different from None, 
        the returned  selection is the conjunction of all of them.
```